src/views/Auth/Login.jsx

```diff
@@ -18,31 +18,40 @@ export default function Login() {
   const handleLogin = (event) => {
     event.preventDefault();
     const loginWasSuccessful = auth.login(formState.email, formState.password);
-
-    // TODO: If login was unsuccessful, set an error with a message
-    // to display to the user that their login failed.
-    //
-    // If login was successful, use the history hook
-    // from React Router to replace the current URL with the URL
-    // we need to redirect to.
-    // See https://v5.reactrouter.com/web/api/history for the appropriate method to use
+    if (loginWasSuccessful) {
+      history.replace(from.pathname);
+    } else {
+      setError('Username or password do not match, please retry.');
+    }
   };
 
+  // TODO: If login was unsuccessful, set an error with a message
+  // to display to the user that their login failed.
+  //
+  // If login was successful, use the history hook
+  // from React Router to replace the current URL with the URL
+  // we need to redirect to.
+  // See https://v5.reactrouter.com/web/api/history for the appropriate method to use
+
   return (
     <>
       <h3>You must log in to view the page at {from.pathname}</h3>
       <form onSubmit={handleLogin} className={styles.loginForm}>
-        <label>Email</label>
+        <label htmlFor="email">Email</label>
         <input
           id="email"
           name="email"
           type="email"
+          value={formState.email}
+          onChange={handleFormChange}
```

Comment:
I was corrected on this - since the handleFormChange function is accepting an `event`, we forgo creating an anonymous function so your method is correct after all! 💯 See, even I'm still learning every day! :)